File: src/main/java/com/arcao/auto/value/parcel/CoordinatesTypeAdapter.java
package com.arcao.auto.value.parcel;

import android.os.Parcel;
import com.arcao.geocaching.api.data.coordinates.Coordinates;
import com.ryanharter.auto.value.parcel.TypeAdapter;

public class CoordinatesTypeAdapter implements TypeAdapter<Coordinates> {
  public Coordinates fromParcel(Parcel in) {
    return Coordinates.create(in.readDouble(), in.readDouble());
  }

  public void toParcel(Coordinates value, Parcel dest) {
    dest.writeDouble(value.latitude());
    dest.writeDouble(value.longitude());
  }
}

File: app/src/main/java/com/arcao/auto/value/parcel/CoordinatesTypeAdapter.kt
package com.arcao.auto.value.parcel

import android.os.Parcel

import com.arcao.geocaching.api.data.coordinates.Coordinates
import com.ryanharter.auto.value.parcel.TypeAdapter

class CoordinatesTypeAdapter : TypeAdapter<Coordinates> {
    override fun fromParcel(src: Parcel): Coordinates =
            Coordinates.create(src.readDouble(), src.readDouble())

    override fun toParcel(value: Coordinates, dest: Parcel) {
        with(dest) {
            writeDouble(value.latitude())
            writeDouble(value.longitude())
        }
    }
}
